### ISSUE TITLE
Split scan - Split button is not disabled after clicking split for group chat split request

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#38130]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.50-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** n/a
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** applause internal team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com.
2. Go to workspace chat.
3. Create a scan split request.
4. Click on the split request while it is scanning.
5. Click Split.
6. Note that the split button is disabled.
7. Go to group chat.
8. Create a scan split request.
9. Click on the split request while it is scanning.
10. Click Split.

## Expected Result:
The split button will be disabled, similar to the behavior for split scan request in workspace chat.

## Actual Result:
The split button is not disabled.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/43996225/f0ba97a9-ebac-4e98-b1f9-07e3f682e253

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kavimuru** [comment]
we think this bug might be related to #vip-split
cc @arielgreen

**Comment by kavimuru** [comment]
@strepanier03 FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Split scan - Split button is not disabled after clicking split for group chat split request
### What is the root cause of that problem?
We disable the button depending on `shouldDisplayMerchantError` [here](https://github.com/Expensify/App/blob/0dc2151378a3fdb4328cb3e0f1e5eb2dbc8e1e7c/src/components/MoneyRequestConfirmationList.js#L528C37-L528C100) . As we can see in video the merchant error is displaying for workspace Split but not displaying for group chat Split. 
### What changes do you think we should make in order to solve the problem?
Use the same condition to disable the button that you used to display the error message above the button. Update [this](https://github.com/Expensify/App/blob/0dc2151378a3fdb4328cb3e0f1e5eb2dbc8e1e7c/src/components/MoneyRequestConfirmationList.js#L528C37-L528C100) condition like below
```
const shouldDisableButton = selectedParticipants.length === 0 || !_.isEmpty(formError);
```
### What alternative solutions did you explore? (Optional)

**Comment by dragnoir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The split button is not disabled when there's a form error

### What is the root cause of that problem?

The QA mentioned that split on group chat has an error because the button is not disabled when there's an error on the form.

I think that's correct, we don't disable buttons when there's an error on the form.

So the issue is within the workspace chat split request. 

### What changes do you think we should make to solve the problem?

We need to remove `shouldDisplayMerchantError` from here
https://github.com/Expensify/App/blob/4ac2dd7d391b53b5b8afc480cae61d8335da3209/src/components/MoneyRequestConfirmationList.js#L528

```js
- const shouldDisableButton = selectedParticipants.length === 0 || shouldDisplayMerchantError;
+ const shouldDisableButton = selectedParticipants.length === 0;
```
This way we keep consistency between all pages. and below are some examples of errors on fields and button not disabled.

![image](https://github.com/Expensify/App/assets/12425932/a8b0d27a-7365-4e57-891e-c4bb8cfe41dc)

![image](https://github.com/Expensify/App/assets/12425932/3e213d7e-83e7-4512-9694-13f9f3ead49a)

![image](https://github.com/Expensify/App/assets/12425932/94150aff-0dac-42fd-948d-bd418ec6d67b)


### What alternative solutions did you explore? (Optional)

We can show on error message that the Merchant is required.

**Comment by dragnoir** [comment]
@shawnborton what do you think? Do we need to disable the form button when there's an error on the form? I noticed that across the app, all our forms show errors and do not disable the button.

**Comment by shawnborton** [comment]
Yeah, I believe our default form behavior does not make form submit buttons disabled. cc @Expensify/design @marcaaron

**Comment by dannymcclain** [comment]
Yup! You're right—it's outlined in our [Forms contributing guide](https://github.com/Expensify/App/blob/eb6213f47b87664f5d61af5c068e3ed25110b63e/contributingGuides/FORMS.md#submit-button-disabling): 

>### Form Submission
>#### Submit Button Disabling
>Submit buttons shall not be disabled or blocked from being pressed in most cases. We will allow the user to submit a form and point them in the right direction if anything needs their attention.
>
>The only time we won’t allow a user to press the submit button is when we have submitted the form and are waiting for a response (e.g. from the API). In this case we will show a loading indicator and additional taps on the submit button will have no effect. This is handled by the Form component and will also ensure that a form cannot be submitted multiple times.

**Comment by strepanier03** [comment]
Does this mean we don't want to make a change because we don't disable the button in most cases? 

Or should I put this external?

**Comment by dragnoir** [comment]
@strepanier03 we need to make a change. The issue is about 2 buttons, one disabled and the other not.
And we were asking if we should disable all OR not allow disable state. (Which button to change)

So yes, please make external, a component should be fixed to have consistency between all the UI.

**Comment by strepanier03** [comment]
Got it, thank you for helping me out @dragnoir 🙌

**Comment by jayeshmangwani** [comment]
@GandalfGwaihir Except the you have used `double Boolean negation` instead lodash in your [proposal](https://github.com/Expensify/App/issues/38130#issuecomment-2000416287) ,how your proposal is different from @shahinyan11 's [proposal](https://github.com/Expensify/App/issues/38130#issuecomment-1991593250) and also can you please elaborate how this will solve the current issue ?

**Comment by jayeshmangwani** [comment]
I have asked [here](https://github.com/Expensify/App/commit/c583248e46a710b27696ad5ca6cba1e7534d2cab#r139894045) why we added the disable when we have merchant error, I think this should be mistake to add disable to Form Button when we have merchant error, but lets wait for the @mountiny @waterim 

@strepanier03 And if we agree to change the expected result like mentioned here, then we should update the below chages in the Github issue description

# Action Performed:

1. Go to staging.new.expensify.com.
2. Go to workspace chat.
3. Create a scan split request.
4. Click on the split request while it is scanning.
5. Click Split.

# Expected Result:
The split button should not be disabled.

# Actual Result:
The split button is disabled.

**Comment by jayeshmangwani** [comment]
>  pls check here #38130 and here #38130

@dragnoir I have checked and I have commented on the original PR why it was added,

**Comment by jayeshmangwani** [comment]
If we decided to not to disable the Form button on the merchant error the we can go with @dragnoir 's [proposal](https://github.com/Expensify/App/issues/38130#issuecomment-1991833010).
 🎀 👀 🎀 C+ reviewed

**Comment by jayeshmangwani** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [x]  [@jayeshmangwani] The PR that introduced the bug has been identified. Link to the PR:https://github.com/Expensify/App/pull/32486
- [x]  [@jayeshmangwani] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:https://github.com/Expensify/App/pull/32486#discussion_r1545805500
- [ ]  [@jayeshmangwani] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A - **this issue may be oversight as we already have mentioned in the Form guidelines that button should not be disabled in most cases**
- [x]  [@jayeshmangwani] Determine if we should create a regression test for this bug.
- [x]  [@jayeshmangwani] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

#  Regression Test Steps

1. Open App & Go to workspace chat.
2. Create a scan split request.
3. Click on the split request while it is scanning.
4. Click Split.
5. Verify that the split button is not disabled.


Do you agree 👍 or 👎 ?

**Comment by jayeshmangwani** [comment]
@strepanier03 This issue is ready for payout.

**Comment by strepanier03** [comment]
Thanks @jayeshmangwani - I'll update the priority label, I was expecting this to flip to Daily and missed it.

**Comment by strepanier03** [comment]
@dragnoir - I just hired you to a job, I'll keep an eye out to pay it when you accept. 

@jayeshmangwani - Paid and closed via Upwork as well. 

Thank you both!

**Comment by strepanier03** [comment]
Saw you accepted so I paid you out as well @dragnoir. 

Thanks again everyone!
